content/docs/user-guide/assets/pipeline/asset-builders.md

```diff
@@ -13,28 +13,147 @@ The Open Asset Import Library supports [many scene formats](https://github.com/a
+
+![Source asset references](/images/user-guide/assets/pipeline/asset_builders/source_asset_references.png)
+
+Source assets and non-asset source files can reference each other in many different ways, and each case may require unique handling at the time of authoring an asset builder.
+
+Source assets and non-asset source files may have these references:
+* source assets by UUID
+* source assets by a path
+* non-asset source files in a scan directory by a path
+* non-asset source files outside a scan directory by a path
+* product assets by asset ID
+* product assets by a path
+
+Path references from source asset and non-asset source files may be defined in these ways:
+* Relative to a scan directory
+* Relative to the asset with the reference
+* Absolute
+* Relative to a directory that may be difficult to infer from the contents or location of the source asset
+
+Source assets come in two forms, relative to how these references work:
+* A format that the builder author can modify and control.
+  * Example: A custom format defined for O3DE, like prefab files.
+* A format that the builder author has no control over.
+  * Example: A format defined outside of O3DE, like FBX files.
+
+When authoring a builder, if the format of the source asset is something the author has control over, then it's useful to manage how the source asset references other files.
+* When referencing a source asset or product asset, it's preferred to reference by UUID or Asset ID, because these will be more stable than a path. Read more about [asset identifiers here.](/docs/user-guide/assets/pipeline/asset-dependencies-and-identifiers/#asset-identifiers)
+
+Resolving paths to find the file on disk is often done within the asset builder. When dealing with path references from the source asset to other files, keep these things in mind:
+* The referenced asset may not be on the same drive as the asset with the reference. This includes scan directories, which can be on different drives. This means relative paths may not be resolvable.
+* Some operating systems have case sensitive paths, some do not. When resolving paths in an asset builder, it's important to do so in a way that will be stable across operating systems.
+* File locations relative to scan directories, and in the asset cache will be stable for all members of a team, but scan directories themselves, and files outside scan directories may not have the same paths for all team members.
+* Path resolving logic should be consistent, and predictable. Content creators will want to understand how paths are resolved when processing assets, so they can better manage their content.
+
+#### References from product assets to other files
+
+![Product asset references](/images/user-guide/assets/pipeline/asset_builders/product_asset_references.png)
+
+Product assets should only have these references:
+* Product assets by asset ID
+* Product assets by a relative path from the asset with the reference
```

Comment:
Maybe we can add a note here to avoid using relative path if possible.

Reply:
Added a note